src/main/java/net/glowstone/block/state/GlowDispenser.java

```diff
@@ -68,17 +90,37 @@ public boolean dispense() {
 
     public int getDispenseSlot() {
         Inventory inv = getInventory();
-
-        int slot = -1;
-        int randomChance = 1;
-
-        for (int i = 0; i < inv.getSize(); i++) {
-            if (inv.getItem(i) != null && random.nextInt(randomChance++) == 0) {
-                slot = i;
+        
+        boolean isEmpty = true;
+        for (ItemStack stack : inv.getContents()) {
+            if (InventoryUtil.isEmpty(stack) == false) isEmpty = false;
+        } 
+        if (isEmpty == false) {
+            int slot = 0;
+            int tries = 0;
+            while (true) {
+                //Gets a random slot and determines if that slot has an item in it
+                slot = random.nextInt(9);
+                if (inv.getItem(slot).getType() != Material.AIR) break;
+                
+                //Eliminates the possibility of an infinite loop, goes in order upon reaching 20 random tries
+                tries += 1;
+                if (tries == 20) {
```

Comment:
Generally, you shouldn't use tries for this. An inventory isn't a black box where we don't know how many slots there are or if any of the have items in it. We can gather all slots with items through inventory methods.